extensions/interactions/FractionInput/FractionInputRulesServiceSpec.js

```diff
@@ -0,0 +1,127 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for Fraction Input rules.
+ */
+
+describe('Fraction Input rules service', function() {
+  beforeEach(module('oppia'));
+
+  var firs = null;
+  beforeEach(inject(function($injector) {
+    firs = $injector.get('fractionInputRulesService');
+  }));
+
+  var createFractionDict = function(wholeNumber, numerator, denominator) {
+    return {
+      isNegative: false,
+      wholeNumber: wholeNumber,
+      numerator: numerator,
+      denominator: denominator
+    }
+  };
+
+  var RULE_INPUT = {
+    f: {
+      isNegative: false,
+      wholeNumber: 1,
+      numerator: 40,
+      denominator: 20
+    }
+  };
+
+  it('should have a correct \'equivalence\' rule', function() {
```

Comment:
Check the case where they're equivalent except for the sign? Ditto below. You could have createPositiveFraction and createNegativeFraction helper functions, if you like.

Reply:
Done.